crates/net/peers/src/node_record.rs

```diff
@@ -196,6 +197,26 @@ impl FromStr for NodeRecord {
     }
 }
 
+impl From<TrustedPeer> for NodeRecord {
+    fn from(trust: TrustedPeer) -> Self {
+        use std::net::{SocketAddr, ToSocketAddrs};
+        let lookup_host = |host: &str| -> Option<SocketAddr> {
+            let mut addrs = host.to_socket_addrs().expect("No IP found");
+            addrs.next()
+        };
+
+        let address = match trust.host {
+            Host::Ipv4(ip) => IpAddr::V4(ip),
+            Host::Ipv6(ip) => IpAddr::V6(ip),
+            Host::Domain(domain) => {
+                let ip = lookup_host(&format!("{domain}:0")).expect("No IP found");
+                ip.ip()
+            }
+        };
+        Self { address, tcp_port: trust.tcp_port, udp_port: trust.udp_port, id: trust.id }
+    }
+}
```

Comment:
this is not want we want, this can panic on domain input.

this would render the TrustedPeer arg format useless.

I think we need to resolve this.

@Rjected I find having to do async resolve during setup slightly annoying. because this now forces us to dod async setup, or we do some block in place which is also not great

please advise

Reply:
Ok, after #9258 is merged, I will use resolve_blocking instead

Reply:
@mattsse @Rjected Please review it again.